plugin/nginx_plus_collectd.py

```diff
@@ -293,6 +297,13 @@ def configure(self, conf):
                 username = node.values[0]
             elif node.key == PASSWORD:
                 password = node.values[0]
+            elif node.key == LEGACY_API:
+                legacy_api = self._str_to_bool(node.values[0])
+            elif node.key == API_VERSION:
+                try:
+                    api_version = int(node.values[0])
+                except ValueError:
+                    print 'Please provide an interger in API version'
```

Comment:
```suggestion
                    print 'APIVersion not configured by user. Using default version of 3.'
```

Also, is this guaranteed to be a `ValueError`?

Reply:
Yes, If a non-integer value is provided as input, then we'll get a `ValueError`.

Reply:
If a user enters a list or another section for the value (invalid input) I believe there could be a `TypeError`:

```int(object())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() argument must be a string, a bytes-like object or a number, not 'object'```

At any rate we are ignoring user input here and I'm not sure this is desirable. If an invalid value is used we should probably raise it as an informative error instead of moving on with the default.

Reply:
Hello @rmfitzpatrick ,

We have added explicit exception handling of the following types for the config option `APIVersion`:

- ValueErrror
- TypeError
- Exception(generic)

Please let us know in case of any queries.

Reply:
This can be done with a single try/except with multiple clauses - `(ValueError, TypeError, Exception)`, but since you are catching all you can just raise the caught type.

Reply:
In the last pushed commit, we have made the change. Now, we are returning the caught type of error with the updated error message.